binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,12 +3,31 @@
 
 import filecmp
 import os
+import re
 import sys
+import typing
 from argparse import ArgumentParser
 
 from ..reporter import exception_reporting
 from ._parser import add_version_argument_to
 
+_keywords_pattern = re.compile('KEYWORDS="(?P<keywords>[^"])"')
+
+
+def _get_relevant_keywords_set_for(ebuild_content: str, accept_keywords: typing.Iterable) -> set:
+    match = _keywords_pattern.search(ebuild_content)
+    if match is None:
+        # if the KEYWORDS variable is not found in the ebuild,
+        # we will assume the ebuild needs to be listed
+        ebuild_keywords = accept_keywords
+    else:
+        ebuild_keywords = match.group('keywords').split(" ")
+
+    if isinstance(accept_keywords, str):
+        accept_keywords = accept_keywords.split(" ")
```

Comment:
The original `accept_keywords` is already assigned further above, so this is coming to later (or too far down). We need to post-process accept keywords anyway, and if that is done elsewhere, the splitting needs to be done elsewhere already as well. Also, our own code uses only one variant of this function so why support two variants in non-API code as of now.